source/agora/common/Hash.d

```diff
@@ -233,3 +203,62 @@ nothrow @nogc @safe unittest
     str.c2 = 'c';
     assert(hashFull(str) == abc_exp);
 }
+
+/*******************************************************************************
+
+    Hashes multiple arguments together
+
+    Params:
+        T = variadic argument types
+        args = the arguments
+
+    Returns:
+        the hash of all the arguments
+
+*******************************************************************************/
+
+public Hash hashMulti (T...)(T args) nothrow @nogc @trusted
+{
+    Hash hash = void;
+    crypto_generichash_state state;
+    crypto_generichash_init(&state, null, 0, Hash.sizeof);
+    scope HashDg dg = (scope const(ubyte)[] data) @trusted
+        => cast(void)crypto_generichash_update(&state, data.ptr, data.length);
+    static foreach (arg; args)
+        hashPart(arg, dg);
```

Comment:
That static foreach can't be `@trusted`

Reply:
Pourquoi?

Reply:
ah

Reply:
I see what you mean.

Reply:
It still is.
You need the rest of the function to be `@trusted`, but `arg` could be anything, including a `system` `opCall`.

Reply:
Indeed. And also hashPart too. I'll fix this in a bit.